sklearn/linear_model/_ridge.py

```diff
@@ -1457,45 +1471,51 @@ def fit(self, X, y, sample_weight=None):
             X, y = _rescale_data(X, y, sample_weight)
             sqrt_sw = np.sqrt(sample_weight)
         else:
-            sqrt_sw = np.ones(X.shape[0], dtype=X.dtype)
+            sample_weight = sqrt_sw = np.ones(X.shape[0], dtype=X.dtype)
+
+        X_mean, *decomposition = decompose(X, y, sqrt_sw)
 
         scorer = check_scoring(self, scoring=self.scoring, allow_none=True)
         error = scorer is None
 
         n_y = 1 if len(y.shape) == 1 else y.shape[1]
-        cv_values = np.zeros((n_samples * n_y, len(self.alphas)),
-                             dtype=X.dtype)
-        C = []
-        X_mean, *decomposition = decompose(X, y, sqrt_sw)
+
+        if self.store_cv_values:
+            self.cv_values_ = np.empty(
+                (n_samples * n_y, len(self.alphas)), dtype=X.dtype)
+
+        best_coef, best_score, best_alpha = None, None, None
+
         for i, alpha in enumerate(self.alphas):
             G_inverse_diag, c = solve(
                 float(alpha), y, sqrt_sw, X_mean, *decomposition)
             if error:
                 squared_errors = (c / G_inverse_diag) ** 2
-                cv_values[:, i] = squared_errors.ravel()
+                if y.ndim == 2:
+                    squared_errors /= sample_weight[:, np.newaxis]
+                else:
+                    squared_errors /= sample_weight
```

Comment:
It is not that easy. There is a test `ridge_sample_weight` which will fail.
Right now it was thought that repeating 3 times a sample will lead to an error 3 times bigger.
Normalizing the sample_weight will not lead to this results

Reply:
this part makes sure that ``RidgeCV()`` is equivalent to ``GridSearchCV(Ridge(), cv=LeaveOneOut())``